modules/sov-ibc/src/rpc/responses.rs

```diff
@@ -0,0 +1,477 @@
+use ibc_core::client::context::client_state::ClientStateCommon;
+use ibc_core::client::types::Height;
+use ibc_core::host::ValidationContext;
+use ibc_query::core::channel::{
+    QueryChannelClientStateRequest, QueryChannelClientStateResponse,
+    QueryChannelConsensusStateRequest, QueryChannelConsensusStateResponse, QueryChannelRequest,
+    QueryChannelResponse, QueryNextSequenceReceiveRequest, QueryNextSequenceReceiveResponse,
+    QueryPacketAcknowledgementRequest, QueryPacketAcknowledgementResponse,
+    QueryPacketCommitmentRequest, QueryPacketCommitmentResponse, QueryPacketReceiptRequest,
+    QueryPacketReceiptResponse,
+};
+use ibc_query::core::client::{
+    IdentifiedClientState, QueryClientStateRequest, QueryClientStateResponse,
+    QueryConsensusStateRequest, QueryConsensusStateResponse, QueryUpgradedClientStateRequest,
+    QueryUpgradedClientStateResponse, QueryUpgradedConsensusStateRequest,
+    QueryUpgradedConsensusStateResponse,
+};
+use ibc_query::core::connection::{
+    QueryClientConnectionsRequest, QueryClientConnectionsResponse,
+    QueryConnectionClientStateRequest, QueryConnectionClientStateResponse,
+    QueryConnectionConsensusStateRequest, QueryConnectionConsensusStateResponse,
+    QueryConnectionRequest, QueryConnectionResponse,
+};
+use jsonrpsee::core::RpcResult;
+use sov_ibc_transfer::to_jsonrpsee_error;
+use sov_modules_api::Spec;
+
+use crate::context::IbcContext;
+use crate::helpers::{WithProof, WithoutProof};
+
+impl<'a, S: Spec> IbcContext<'a, S> {
+    /// Determines the query height to use for the given request. If the query
+    /// height is not provided, it queries the host for the current height.
+    pub(super) fn determine_query_height(&self, query_height: Option<Height>) -> RpcResult<Height> {
+        let height = match query_height {
+            Some(height) => height,
```

Comment:
do we still need this method? The way I see it, this height must always be `self.host_height()`.

Reply:
ac64484